### PR TITLE
Add docker setup resources to the kafka connect repo

### PR DESCRIPTION
**Pull request by sfc-gh-rcheng**
The goal of this PR is to introduce a docker-setup folder that will have all the resources needed to setup docker. Previously setup required pulling @sfc-gh-japatel branch.

Main changes: 
- Created m1 specific yml file
- Added template connector json files
- Created folder for user to customize their json files without git tracking
- Onboarding document references this folder

Jay's branch: https://github.com/sfc-gh-japatel/snowflake-kafka-docker/tree/snowpipe_streaming_demo
Current docker documentation: https://snowflakecomputing.atlassian.net/wiki/spaces/EN/pages/2466252387/Snowflake+Kafka+Connector+In+Docker+Container
Previous docker documentation: https://snowflakecomputing.atlassian.net/wiki/pages/viewpage.action?pageId=2466252387&pageVersion=10

### CODE REVIEW

File: docker-setup/snowflake-kafka-docker/kafka-connect.yml [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      - kafka
+    ports:
+      - "8083:8083"
+      - 42514:42514/udp
+      - 5060:5060 # KAFKA_DEBUG port
+      - 9876:9876 # JMX port
+      - 8085:8085
+    volumes:
+      - ./jars:/etc/kafka-connect/jars/ # if you want your custom jar, put custom jar in ./jars directory
+      - ./prometheus:/opt/jmx-exporter/
```
Comment (sfc-gh-japatel) [inline docker-setup/snowflake-kafka-docker/kafka-connect.yml:69]: lets comment this line

File: docker-setup/snowflake-kafka-docker/kafka-connect.yml [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      CONNECT_INTERNAL_KEY_CONVERTER: "org.apache.kafka.connect.json.JsonConverter"
+      CONNECT_INTERNAL_VALUE_CONVERTER: "org.apache.kafka.connect.json.JsonConverter"
+      # CLASSPATH required due to CC-2422
+      CLASSPATH: /usr/share/java/monitoring-interceptors/monitoring-interceptors-7.1.0.jar
+      # CONNECT_PRODUCER_INTERCEPTOR_CLASSES: "io.confluent.monitoring.clients.interceptor.MonitoringProducerInterceptor"
+      # CONNECT_CONSUMER_INTERCEPTOR_CLASSES: "io.confluent.monitoring.clients.interceptor.MonitoringConsumerInterceptor"
+      CONNECT_PLUGIN_PATH: "/usr/share/java,/usr/share/confluent-hub-components,/etc/kafka-connect/jars"
+      # CONNECT_PLUGIN_PATH: "/usr/share/java,/usr/share/confluent-hub-components"
+      JMX_PORT: 9876
+      KAFKA_JMX_OPTS: "-javaagent:/opt/jmx-exporter/jmx_prometheus_javaagent-0.16.0.jar=8085:/opt/jmx-exporter/kafka-connect-pattern.yml -Djava.rmi.server.hostname=0.0.0.0 -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.port=9876 -Dcom.sun.management.jmxremote.rmi.port=9876 -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false"
```
Comment (sfc-gh-japatel) [inline docker-setup/snowflake-kafka-docker/kafka-connect.yml:95]: lets comment this line too

File: docker-setup/snowflake-kafka-docker/kafka-connect.yml [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+      CONNECT_LOG4J_LOGGERS: org.apache.zookeeper=ERROR,org.I0Itec.zkclient=ERROR,org.reflections=ERROR,com.snowflake=DEBUG
+      KAFKA_DEBUG: true
+      JAVA_DEBUG_PORT: 0.0.0.0:5060
+    command:
+      - bash
+      - -c
+      - |
+        # echo "Installing Snowflake kafka connector"
+        # if jar is available on confluent hub, uncomment below line
+        # confluent-hub install --no-prompt snowflakeinc/snowflake-kafka-connector:1.8.0
```
Comment (sfc-gh-japatel) [inline docker-setup/snowflake-kafka-docker/kafka-connect.yml:105]: And uncomment this line to start with SF Connector 
Also, lets upgrade it to 1.8.1

**Review by sfc-gh-japatel — APPROVED** [review]
minor comments, but looks good to me. Thanks for setting up guidelines for M1 macs and improving existing documentation.

File: docker-setup/restart-SF-connector-task.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+       sleep 100
+       ((count=count+1))
+    done
+    exec "$@"
+fi
+
+
+
+
+
```
Comment (sfc-gh-tzhang) [inline docker-setup/restart-SF-connector-task.sh:33]: nit: remove empty lines?

File: docker-setup/README.md [hunk 5]
```diff
@@ -0,0 +1,2 @@
+# Snowflake Sink Connector Docker 
+Onboarding instructions here: [Snowflake Kafka Connector In Docker Container](https://snowflakecomputing.atlassian.net/wiki/spaces/EN/pages/2466252387/Snowflake+Kafka+Connector+In+Docker+Container)
```
Comment (sfc-gh-tzhang) [inline docker-setup/README.md:2]: do you want to update the doc as well to reflect the process changes after this is merged?

File: docker-setup/snowflake-kafka-docker/kafka-connect.yml [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+      CONNECT_INTERNAL_KEY_CONVERTER: "org.apache.kafka.connect.json.JsonConverter"
+      CONNECT_INTERNAL_VALUE_CONVERTER: "org.apache.kafka.connect.json.JsonConverter"
+      # CLASSPATH required due to CC-2422
+      CLASSPATH: /usr/share/java/monitoring-interceptors/monitoring-interceptors-7.1.0.jar
+      # CONNECT_PRODUCER_INTERCEPTOR_CLASSES: "io.confluent.monitoring.clients.interceptor.MonitoringProducerInterceptor"
+      # CONNECT_CONSUMER_INTERCEPTOR_CLASSES: "io.confluent.monitoring.clients.interceptor.MonitoringConsumerInterceptor"
+      CONNECT_PLUGIN_PATH: "/usr/share/java,/usr/share/confluent-hub-components,/etc/kafka-connect/jars"
+      # CONNECT_PLUGIN_PATH: "/usr/share/java,/usr/share/confluent-hub-components"
+      JMX_PORT: 9876
+      #KAFKA_JMX_OPTS: "-javaagent:/opt/jmx-exporter/jmx_prometheus_javaagent-0.16.0.jar=8085:/opt/jmx-exporter/kafka-connect-pattern.yml -Djava.rmi.server.hostname=0.0.0.0 -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.port=9876 -Dcom.sun.management.jmxremote.rmi.port=9876 -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false"
```
Comment (sfc-gh-tzhang) [inline docker-setup/snowflake-kafka-docker/kafka-connect.yml:95]: IIRC, Zekai had issue about this as well, I think we need this for JMX metrics? @sfc-gh-japatel

File: docker-setup/README.md (hunk 5 above)
Comment (sfc-gh-rcheng) [reply docker-setup/README.md:2]: The dock has already been updated! 
https://snowflakecomputing.atlassian.net/wiki/spaces/EN/pages/2466252387/Snowflake+Kafka+Connector+In+Docker+Container

**Review by sfc-gh-tjones — APPROVED** [review]

File: docker-setup/snowflake-kafka-docker/kafka-connect.yml (hunk 6 above)
Comment (sfc-gh-japatel) [reply docker-setup/snowflake-kafka-docker/kafka-connect.yml:95]: We dont need jmx metrics unless we plan to expose it on prometheus and grafanna. 

Just the JMX port 9876 is enough! This line is specific for prometheus jmx java agent

File: docker-setup/snowflake-kafka-docker/m1-kafka-connect.yml [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+      - "9021:9021"
+    environment:
+      CONTROL_CENTER_BOOTSTRAP_SERVERS: 'kafka:29092'
+      CONTROL_CENTER_ZOOKEEPER_CONNECT: 'zookeeper:2181'
+      CONTROL_CENTER_CONNECT_CLUSTER: 'connect:8083'
+      CONTROL_CENTER_REPLICATION_FACTOR: 1
+      CONTROL_CENTER_INTERNAL_TOPICS_PARTITIONS: 1
+      CONTROL_CENTER_CONNECT_HEALTHCHECK_ENDPOINT: '/connectors'
+      #CONTROL_CENTER_LOG4J_ROOT_LOGLEVEL: "DEBUG" # uncomment for debug logs
+      PORT: 9021
```
Comment (sfc-gh-japatel) [inline docker-setup/snowflake-kafka-docker/m1-kafka-connect.yml:126]: Should we bring up a discussion in confluent sync that control center is not working on m1 machines?

**Review by sfc-gh-japatel — APPROVED** [review]
lgtm

File: docker-setup/snowflake-kafka-docker/m1-kafka-connect.yml (hunk 7 above)
Comment (sfc-gh-rcheng) [reply docker-setup/snowflake-kafka-docker/m1-kafka-connect.yml:126]: I mentioned it during the kafka connector sync and raised an issue on the confluent control center github: https://github.com/confluentinc/control-center-images/issues/55

**Pull request merged**